### PR TITLE
nixos-option: init at <unversioned>

### PR DESCRIPTION
**Pull request by rycee**
###### Motivation for this change

Make nixos-option a proper package so that it is more easily accessible to code outside the nixos directory. Also, this should hopefully make the package be available in the binary cache for Darwin.

This change was motivated by the use of nixos-option in Home Manager.

Note, I arbitrarily set the version to "0.0.20191217" based on the date when the code was last changed. If anybody has a different preferred version number, let me know and I will update the PR.

Btw, it would be great to have this backported to 21.05 but I don't insist on it.

CC @chkno

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by roberth — COMMENTED** [review]
This is a good idea.

I've done something [similar](https://github.com/NixOS/nixpkgs/pull/122784) with `nixos-install-tools`. That ran into a (solved) complication with tooling rebuilding it because of a dependency on the manual (and all options). Despite its name, `nixos-option` doesn't have such a build dependency, so that problem won't apply here.

File: pkgs/tools/misc/nixos-option/default.nix [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {lib, stdenv, boost, cmake, pkg-config, nix, ... }:
+
 stdenv.mkDerivation rec {
-  name = "nixos-option";
+  pname = "nixos-option";
+  version = "0.0.20191217";
```
[the remark this answers is not in the document]
Comment (rycee) [reply pkgs/tools/misc/nixos-option/default.nix:5]: No need, we already use nixos-option without version number: https://github.com/nix-community/home-manager/blob/06ee8ec8dfc4554c374052cd7b7083765748af99/home-manager/default.nix#L14-L15

So I'd be happy to remove the version number.

**Review by roberth — APPROVED** [review]
cc @chkno

File: pkgs/tools/misc/nixos-option/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (rycee) [reply pkgs/tools/misc/nixos-option/default.nix]: Removed the version number.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129696` at ce6f17f9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nixos-option</li>
    <li>tests.trivial</li>
  </ul>
</details>

**Comment by marsam** [comment]
This breaks [nix-darwin](https://github.com/LnL7/nix-darwin)

```
$ darwin-rebuild build
building the system configuration...
error: getting status of '/Users/marsam/code/nix-config/nixpkgs/nixos/modules/installer/tools/nixos-option': No such file or directory
(use '--show-trace' to show detailed location information)
```

<details>
<summary>darwin-rebuild build --show-trace</summary>

```
$ darwin-rebuild build --show-trace
building the system configuration...
error: while evaluating the attribute 'activationScript' of the derivation 'darwin-system-21.11.git.ce6f17f+darwin3.007d700' at /Users/marsam/code/nix-config/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:203:11:
while evaluating the attribute 'activationScripts.script.text' at undefined position:
while evaluating 'g' at /Users/marsam/code/nix-config/nixpkgs/lib/attrsets.nix:298:19, called from undefined position:
while evaluating anonymous function at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:140:72, called from /Users/marsam/code/nix-config/nixpkgs/lib/attrsets.nix:301:20:
while evaluating the attribute 'value' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:525:9:
while evaluating the option `system.activationScripts.script.text':
while evaluating the attribute 'mergedValue' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:557:5:
while evaluating the attribute 'values' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:551:9:
while evaluating the attribute 'values' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:650:7:
while evaluating anonymous function at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:537:28, called from /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:537:17:
while evaluating definitions from `/Users/marsam/code/nix-config/darwin/modules/system/activation-scripts.nix':
while evaluating 'dischargeProperties' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:609:25, called from /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:538:137:
while evaluating the attribute 'value' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:392:44:
while evaluating the attribute 'activationScripts.applications.text' at undefined position:
while evaluating 'g' at /Users/marsam/code/nix-config/nixpkgs/lib/attrsets.nix:298:19, called from undefined position:
while evaluating anonymous function at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:140:72, called from /Users/marsam/code/nix-config/nixpkgs/lib/attrsets.nix:301:20:
while evaluating the attribute 'value' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:525:9:
while evaluating the option `system.activationScripts.applications.text':
while evaluating the attribute 'mergedValue' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:557:5:
while evaluating the attribute 'values' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:551:9:
while evaluating the attribute 'values' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:650:7:
while evaluating anonymous function at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:537:28, called from /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:537:17:
while evaluating definitions from `/Users/marsam/code/nix-config/darwin/modules/system/applications.nix':
while evaluating 'dischargeProperties' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:609:25, called from /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:538:137:
while evaluating the attribute 'value' at /Users/marsam/code/nix-config/nixpkgs/lib/modules.nix:392:44:
while evaluating the attribute 'passAsFile' of the derivation 'system-applications' at /Users/marsam/code/nix-config/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:203:11:
while evaluating the attribute 'passAsFile' at /Users/marsam/code/nix-config/nixpkgs/pkgs/build-support/buildenv/default.nix:77:5:
while evaluating the attribute 'buildCommand' of the derivation 'home-manager' at /Users/marsam/code/nix-config/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:203:11:
while evaluating 'callPackageWith' at /Users/marsam/code/nix-config/nixpkgs/lib/customisation.nix:117:35, called from /Users/marsam/code/nix-config/home-manager/home-manager/default.nix:15:5:
while evaluating 'makeOverridable' at /Users/marsam/code/nix-config/nixpkgs/lib/customisation.nix:67:24, called from /Users/marsam/code/nix-config/nixpkgs/lib/customisation.nix:121:8:
getting status of '/Users/marsam/code/nix-config/nixpkgs/nixos/modules/installer/tools/nixos-option': No such file or directory```
</details>

**Comment by rycee** [comment]
@marsam This is because you pull in Home Manager, which depends on nixos-option. I don't think it should break nix-darwin as such.

In any case, Home Manager has been updated to use the package if available. If you do an update I believe it should be build now.

**Comment by zowoq** [comment]
Seems this was only being built on linux so I've opened https://github.com/NixOS/nixpkgs/pull/132970.

**Pull request merged**